components/checkbox/index.css

```diff
@@ -24,9 +157,299 @@ governing permissions and limitations under the License.
+    &.is-invalid .spectrum-Checkbox-input.focus-ring:checked + .spectrum-Checkbox-box {
+      &:before {
+        border-color: var(--highcontrast-checkbox-color-focus, var(--mod-checkbox-invalid-color-focus, var(--spectrum-checkbox-invalid-color-focus)));
+      }
+    }
+
+    /* Invalid Hover States */
+    &.is-invalid:hover {
+      &.is-indeterminate .spectrum-Checkbox-box,
+      .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--highcontrast-checkbox-color-hover, var(--mod-checkbox-invalid-color-hover, var(--spectrum-checkbox-invalid-color-hover)));
+        }
+      }
+    }
+
+    /* Selected and Indeterminate Hover States */
+    &:hover {
+      &.is-indeterminate .spectrum-Checkbox-box,
+      .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--highcontrast-checkbox-highlight-color-hover, var(--mod-checkbox-emphasized-color-hover, var(--spectrum-checkbox-emphasized-color-hover)));
+        }
+      }
+    }
+
+    /* Selected and Indeterminate Active States */
+    &:active {
+      &.is-indeterminate .spectrum-Checkbox-box,
+      .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--highcontrast-checkbox-highlight-color-default, var(--mod-checkbox-emphasized-color-down, var(--spectrum-checkbox-emphasized-color-down)));
+        }
+      }
+    }
+
+    /* Invalid Active State */
+    &.is-invalid:active {
+      .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box,
+      .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--highcontrast-checkbox-highlight-color-default, var(--mod-checkbox-control-invalid-color-down, var(--spectrum-checkbox-invalid-color-down)));
+        }
+      }
+    }
+
+    /* Focus State */
+    &:focus {
+      .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box,
+      .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--highcontrast-checkbox-color-focus, var(--mod-checkbox-control-color-focus, var(--spectrum-checkbox-control-color-focus)));
+        }
+      }
+    }
+  }
+}
+
+.spectrum-Checkbox-label {
+  text-align: start;
+  margin-inline-start: var(--spectrum-checkbox-text-to-control);
+  margin-block-start: var(--spectrum-checkbox-top-to-text);
+
+  font-size: var(--spectrum-checkbox-font-size);
+
+  /* TODO Use font-weight and font-style tokens */
+  /* font-weight: var(--spectrum-checkbox-text-font-weight); */
+  /* font-style: var(--spectrum-checkbox-text-font-style); */
+
+  transition: color var(--spectrum-checkbox-animation-duration) ease-in-out;
+
+  line-height: var(--spectrum-checkbox-line-height);
+
+  &:lang(js),
+  &:lang(zh),
+  &:lang(ko) {
+    line-height: var(--spectrum-checkbox-CJK-line-height, var(--spectrum-checkbox-line-height));
+  }
 }
 
 .spectrum-Checkbox-input {
+  color: var(--spectrum-checkbox-control-color-default);
+
```

Comment:
Can you add the `--mod-` custom properties to each of the custom properties used here?